refinery/analysis_manager/tests.py

```diff
@@ -0,0 +1,398 @@
+import json
+import uuid
+
+import mock
+from django.contrib.auth.models import User
+from django.http import HttpResponseBadRequest, HttpResponseNotAllowed
+from django.test import RequestFactory, TestCase
+from guardian.utils import get_anonymous_user
+
+from analysis_manager.tasks import run_analysis
+from analysis_manager.utils import (create_analysis,
+                                    fetch_objects_required_for_analysis,
+                                    validate_analysis_config)
+from analysis_manager.views import run
+from core.models import Analysis, Workflow, WorkflowEngine
+from factory_boy.utils import (create_dataset_with_necessary_models,
+                               make_analyses_with_single_dataset)
+from galaxy_connector.models import Instance
+
+
+class AnalysisConfigTests(TestCase):
+    """
+    Test for validation of the two legacy ways of launching Analyses (
+    using NodeSets and NodeRelationships) as well as for the new Tool-based
+    approach
+    """
+    def test_valid_nodeset_analysis_config(self):
+        validate_analysis_config(
```

Comment:
add comment about why there's no assert: "If I did assert that no RuntimeError was raised, then later down the line if `validate_analysis_config` changes and raises some new Exception it could fall through the cracks"